Add new Getting Started Guide (#1340)
terramate-io/terramate: docs/cli/getting-started/index.md

::: tip
Join the ***[Terramate community](https://terramate.io/discord)*** on Discord if you have questions, need help, or to
join a fun community focussing on all things Infrastructure as Code. Come aboard ⛵️🏴‍☠️

👉 [https://terramate.io/discord](https://terramate.io/discord)
:::

## Introduction

Terramate works best when used inside a git repository. This repository does not need to be remote or synchronized to
any VCS Platform such as GitHub, GitLab, or BitBucket - a local repository is a great start to test things.

In this Quickstart Guide, we will do the following:

- Initialize a new git repository
- Create two stacks `first` and `second` and learn about Terramate's [Change Detection](../change-detection/index.md)
- Generate a Terraform Local Backend configuration in all stacks at once from a central place
- Create a plain Terraform Null Resource in the `second` stack
- Preview and apply the Null Resource using Terraform while leveraging [Change Detection](../change-detection/index.md)
and [Orchestration](../orchestration/index.md) features of Terramate.

## Prerequisites

You will need the following tooling installed:

> - Any recent version of the `git` command line tool

Reviewer: ```suggestion
- A recent version of the `git` command line tool
```

Author: resolved

> - Any current version of [Terraform](https://developer.hashicorp.com/terraform/install)

Reviewer: ```suggestion
- A recent version of [Terraform](https://developer.hashicorp.com/terraform/install)
```

Author: resolved

For details, please see [code generation best practices](../code-generation/index.md#best-practices).
:::

## Run Terraform in both stacks

The stacks created in the previous sections both represent isolated Terraform environments. To make them functional,
we must run `terraform init` in both. This is where the Terramate Orchestration comes in handy, which allows us to
[run commands in stacks](../orchestration/run-commands-in-stacks.md) with the [run](../cmdline/run.md) command.
As mentioned before, Terramate is not limited to orchestrating Terraform but can run any command.

But before we can start, we need to prepare git to ignore terraform temporary files by adding a `.gitignore` file, which is
located in the root directory of our repository:

```sh
# NOTE: 
# You might not want to add state and lock file here
# This is just convenient when running the quickstart guide
cat <<EOF >.gitignore
.terraform
.terraform.lock.hcl
terraform.tfstate
EOF

git add .gitignore
git commit -m 'Add .gitignore'
```

Now let’s initialize Terraform in all stacks:

::: code-group
```sh [shell]
terramate run terraform init
```

> ```sh [output]

Reviewer: The output looks a bit strange (the first word of each line is highlighted, see screenshot below). We could drop the `sh` from the code block to not use any colors in the output? 🤔 

![image](https://github.com/terramate-io/terramate/assets/92498/eaba71cc-df92-4f23-88d8-95dadb99d43d)

Author: changed this to text output instead - very good catch!

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # null_resource.quickstart will be created
  + resource "null_resource" "quickstart" {
      + id = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
null_resource.quickstart: Creating...
null_resource.quickstart: Creation complete after 0s [id=6372471593458417750]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
:::

When running `terraform plan` again, we expect no changes to be planned anymore:

::: code-group
```sh [shell]
terramate run --changed terraform plan
```

```sh [output]
terramate: Entering stack in /stacks/second
terramate: Executing command "terraform plan"
null_resource.quickstart: Refreshing state... [id=6372471593458417750]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

```
:::

## Conclusion

We hope this tutorial has helped you grasp the basics of Terramate.

> Here's a summary of what we learned in just 5 minutes:

Reviewer: ```suggestion
Here's a summary of what we learned:
```
